Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs

```diff
@@ -1082,7 +1085,7 @@ protected override void Dispose(bool disposing) {
 
                 _recreatingAnalyzer.Dispose();
             }
-
+            _isDisposed = true;
```

Comment:
Maybe it would be better to put this at the top before all the other things are removed? (In case one of them throws an exception.

Then you can also check if it's already disposed or not and throw an exception if it is.